### PR TITLE
Update .NET SDK to `6.0.100-rc.2.21424.34`

### PR DESCRIPTION
**Pull request by dougbu**

### CODE REVIEW

**Comment by captainsafia** [comment]
Suppressed the restore issues per offline discussion. Looks like there is a build error now:

> src/Shared/PropertyHelper/PropertyHelper.cs(288,17): error CS8969: (NETCORE_ENGINEERING_TELEMETRY=Build) Type 'Func<object, object?>' cannot be used in this context because it cannot be represented in metadata.

Looking...

File: eng/Workarounds.props [hunk 1]
```diff
@@ -25,6 +25,12 @@
     <UseSharedCompilation>true</UseSharedCompilation>
   </PropertyGroup>
 
+  <!-- Workaround for RC2 SDK update. -->
```
Comment (dougbu) [inline eng/Workarounds.props:28]:
```suggestion
  <!-- Workaround continued use of netcoreapp2.1. -->
```

File: eng/Workarounds.props [hunk 2]
```diff
@@ -25,6 +25,12 @@
...
+  <PropertyGroup>
+    <NoWarn>NETSDK1138</NoWarn>
+    <NoWarn>CS8969</NoWarn>
```
Comment (dougbu) [inline eng/Workarounds.props:31]:
```suggestion
    <NoWarn>$(NoWarn);NETSDK1138;CS8969</NoWarn>
```

**Review by dougbu — COMMENTED** [review]
Couple of updates&hellip;

**Pull request merged**